src/Metadata/Driver/TypedPropertiesDriver.php

```diff
@@ -102,6 +102,10 @@ public function loadMetadataForClass(ReflectionClass $class): ?ClassMetadata
         foreach ($classMetadata->propertyMetadata as $propertyMetadata) {
             // If the inner driver provides a type, don't guess anymore.
             if ($propertyMetadata->type) {
+                if ('union' === $propertyMetadata->type['name']) {
+                    $propertyMetadata->setType($this->reorderTypes($propertyMetadata->type));
```

Comment:
you could potentially encode in the `union` type already the info regarding the discriminator field and map.
advantage of it is that it would be done only once at compile cache metadata time vs at runtime as done now.
it should be much faster.


```php
#[UnionDiscriminator(
  field: 'objectType', 
  map: [
   'author' => 'JMS\Serializer\Tests\Fixtures\MappedDiscriminatedAuthor', 
   'comment' => 'JMS\Serializer\Tests\Fixtures\MappedDiscriminatedComment'
  ]
)]
```
can be translated to 

```php

[
  'type' => 'union',
  'params' => [
      'objectType',
       [
        'author' => 'JMS\Serializer\Tests\Fixtures\MappedDiscriminatedAuthor', 
        'comment' => 'JMS\Serializer\Tests\Fixtures\MappedDiscriminatedComment'
       ]

  ]
]
```

in this way you have all the data ready in the `UnionHandler` and no need to fetch metadata each time


do you see it as doable approach?